modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2TokenExchangeGrantTypeTestCase.java

```diff
@@ -157,30 +176,238 @@ public void endTest() throws Exception {
+
         JSONObject responseObject = sendPOSTMessage(PRIMARY_IS_TOKEN_ENDPOINT, primaryISClientID, primaryISClientSecret,
                 postParameters);
         String exchangedToken = responseObject.get("access_token").toString();
+        Assert.assertNotNull(exchangedToken, "Access token is null.");
+        Assert.assertEquals(getTokenSubject(exchangedToken), secondaryISUserId, "Subject of the exchanged " +
+                "token should be the same as the subject of the subject token.");
+    }
 
+    private void AssertExchangedTokenForLocalUser(List<NameValuePair> postParameters) throws Exception {
+
+        JSONObject responseObject = sendPOSTMessage(PRIMARY_IS_TOKEN_ENDPOINT, primaryISClientID, primaryISClientSecret,
+                postParameters);
+        String exchangedToken = responseObject.get("access_token").toString();
         Assert.assertNotNull(exchangedToken, "Access token is null.");
+        Assert.assertEquals(getTokenSubject(exchangedToken), primaryISUserId, "Subject of the exchanged " +
+                "token should be the same as the subject of the matching local user.");
+    }
+
+    private void AssertUnauthorizedResponse(List<NameValuePair> postParameters) throws Exception {
+
+        HttpPost httpPost = new HttpPost(PRIMARY_IS_TOKEN_ENDPOINT);
+        httpPost.setHeader("Authorization", "Basic " + getBase64EncodedString(primaryISClientID, primaryISClientSecret));
+        httpPost.setHeader("Content-Type", "application/x-www-form-urlencoded");
+        httpPost.setEntity(new UrlEncodedFormEntity(postParameters));
+        HttpResponse response = client.execute(httpPost);
+        Assert.assertEquals(response.getStatusLine().getStatusCode(), 401, "401 response expected but got: " + response
+                .getStatusLine().getStatusCode());
+    }
+
+    private String getTokenSubject(String token) throws Exception {
+
+        SignedJWT signedJWT;
+        try {
+            signedJWT = SignedJWT.parse(token);
+        } catch (ParseException e) {
+            throw new Exception("Error while parsing the JWT", e);
+        }
+
+        return signedJWT.getJWTClaimsSet().getSubject();
+    }
+
+    private void createUser(int portOffset) throws Exception {
+
+        List<ClaimValue> claimValues = new ArrayList<>();
+        ClaimValue claimValue = new ClaimValue();
+        claimValue.setClaimURI(EMAIL_CLAIM_URI);
+        claimValue.setValue(NEW_USER_EMAIL);
+        claimValues.add(claimValue);
+        super.addUser(portOffset, OAuth2TokenExchangeGrantTypeTestCase.NEW_USER_USERNAME,
+                OAuth2TokenExchangeGrantTypeTestCase.NEW_USER_PASSWORD, null, DEFAULT_PROFILE, claimValues.toArray(new ClaimValue[0]));
+        HashSet<String> users = super.getUserList(PORT_OFFSET_1);
+        Assert.assertTrue(users.contains(OAuth2TokenExchangeGrantTypeTestCase.NEW_USER_USERNAME), "User creation failed in IS: " +
```

Comment:
we should not have asserts outside the tests. Only the methods annotated as Test methods should have Asserts. Please refactor accordingly in all applicable places